impress-org/givewp
# The Onboarding Wizard passes accessibility standards

Resolves #4957

## Description



Using the Axe Accessibility browser tool, I stepped through the Onboarding Wizard and made changes to resolve accessibility issue

### Overview
- Updated each step's main header from `h2` to `h1`, as well as any additional headers levels.
- Added an index to each instance of the `< Checkmark />` and `<GradientChevronIcon />` components so that the `id` attributes of each are unique.
- Added `screen-reader-text` to the Introduction so that the top level header to accessible, which now reads "Welcome to GiveWP", with "GiveWP" hidden as screen reader text. The top level header communicates the title visually, but isn't accessible.
- Updated the "dismiss" link color to increase contrast (now matches the Setup Page).
- Wrapped the "your-cause" `<SelectInput />` component in a `<label />` and screen reader text. The section header communicates the label visually, but isn't accessible.
- Updated the "add-on" badge to increase color contrast (now matches the Setup Page).
- Added the "navigation" role to the `<StepNavigation />` component.
- Added the "main" role to the `<Step />` component, which wraps each step's contents.

## Needs discussion
- Removed the `<Wizard />`'s effect for setting `tabindex`. The WCAG doesn't allow for the `tabindex` to be above zero and states that content should be in the correct order to avoid the appearance of "missed" elements.

### Other changes
- Replaced the `<CardInput />`'s `selected` style from a `border` to a `box-shadow` to prevent movement in the contents when toggled.

## Pre-review Checklist




## Testing Instructions



- git origin fetch
- git checkout `refactor/onboarding-accessibility-4957`
- npm install && npm run dev
- `/wp-admin/?page=give-onboarding-wizard`

Note that the color contrast of the buttons and step-links don't pass the axe accessibility check, but I've addressed the _spirit_ of the issue instead of the _letter_ of the tool's check, see https://github.com/impress-org/givewp/pull/4958#issuecomment-665140278.

## User Documentation
